### ISSUE TITLE
Latest smallest reproducible error with Dask (GITHUB pull) and fastparquet 0.1.3

### ISSUE THREAD

**Issue by apiszcz** [dask/fastparquet#236]
```python
import sys
import dask
from dask import dataframe as dd
import fastparquet as fp
import pandas as pd
import numpy as np
print('dask',dask.__version__,'fastparquet',fp.__version__)
df=pd.DataFrame(np.random.randint(0,100,(5,3)),columns=list('abc'))
fp.write('__fpdasktest.parq',df,file_scheme='hive',partition_on=['a'],compression='SNAPPY')

df2=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
print(df,'\n',df2)

from dask154 import dataframe as dd
df3=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
print(df3)

```

    dask 0.15.2 fastparquet 0.1.2
        a   b   c
    0  62  42  42
    1  73  98  27
    2  43  30  91
    3  27  58   4
    4  56  72  20 
         b   c   a
    0  72  20  56
    0  30  91  43
    0  58   4  27
    0  42  42  62
    0  98  27  73
    

    ---------------------------------------------------------------------------

    TypeError                                 Traceback (most recent call last)

    <ipython-input-22-9c09f1d37407> in <module>()
         13 
         14 from dask154 import dataframe as dd
    ---> 15 df3=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
         16 print(df3)
    

    C:\g\via\lib\dask154\dataframe\io\parquet.py in read_parquet(path, columns, filters, categories, index, storage_options, engine)
        291         return _read_fastparquet(fs, paths, file_opener, columns=columns,
        292                                  filters=filters,
    --> 293                                  categories=categories, index=index)
        294     else:
        295         return _read_pyarrow(fs, paths, file_opener, columns=columns,
    

    C:\g\via\lib\dask154\dataframe\io\parquet.py in _read_fastparquet(fs, paths, myopen, columns, filters, categories, index, storage_options)
         61     name = 'read-parquet-' + tokenize(pf, columns, categories)
         62 
    ---> 63     rgs = [rg for rg in pf.row_groups if
         64            not(fastparquet.api.filter_out_stats(rg, filters, pf.schema)) and
         65            not(fastparquet.api.filter_out_cats(rg, filters))]
    

    C:\g\via\lib\dask154\dataframe\io\parquet.py in <listcomp>(.0)
         63     rgs = [rg for rg in pf.row_groups if
         64            not(fastparquet.api.filter_out_stats(rg, filters, pf.schema)) and
    ---> 65            not(fastparquet.api.filter_out_cats(rg, filters))]
         66 
         67     if index is False:
    

    C:\g\via\lib\fastparquet\api.py in filter_out_cats(rg, filters, sep)
        710                 v0 = val_to_num(v)
        711 
    --> 712             out = filter_val(op, val, v0, v0)
        713             if out is True:
        714                 return True
    

    C:\g\via\lib\fastparquet\api.py in filter_val(op, val, vmin, vmax)
        736         if op in ['==', '>='] and val > vmax:
        737             return True
    --> 738         if op == '>' and val >= vmax:
        739             return True
        740         if op == 'in' and min(val) > vmax:
    

    TypeError: unorderable types: int() >= str()

**Comment by apiszcz** [comment]
Here is a similar error with fastparquet 0.1.3



```python
import sys
import dask
from dask import dataframe as dd
import fastparquet as fp
import pandas as pd
import numpy as np
print('dask',dask.__version__,'fastparquet',fp.__version__)
df=pd.DataFrame(np.random.randint(0,100,(5,3)),columns=list('abc'))
fp.write('__fpdasktest.parq',df,file_scheme='hive',partition_on=['a'],compression='SNAPPY')

df2=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
print(df,'\n',df2)

from dask154 import dataframe as dd
df3=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
print(df3)

```

    dask 0.15.2 fastparquet 0.1.3
        a   b   c
    0  52  90  97
    1   3  52  97
    2  88  43  99
    3  53   2  38
    4  11  81  60 
         b   c   a
    0  81  60  11
    0  43  99  88
    0  90  97  52
    0   2  38  53
    


    ---------------------------------------------------------------------------

    TypeError                                 Traceback (most recent call last)

    <ipython-input-1-9c09f1d37407> in <module>()
         13 
         14 from dask154 import dataframe as dd
    ---> 15 df3=dd.read_parquet('__fpdasktest.parq', filters=[('a','>',10)]).compute()
         16 print(df3)
    

    C:\g\via\lib\dask154\dataframe\io\parquet.py in read_parquet(path, columns, filters, categories, index, storage_options, engine)
        291         return _read_fastparquet(fs, paths, file_opener, columns=columns,
        292                                  filters=filters,
    --> 293                                  categories=categories, index=index)
        294     else:
        295         return _read_pyarrow(fs, paths, file_opener, columns=columns,
    

    C:\g\via\lib\dask154\dataframe\io\parquet.py in _read_fastparquet(fs, paths, myopen, columns, filters, categories, index, storage_options)
         61     name = 'read-parquet-' + tokenize(pf, columns, categories)
         62 
    ---> 63     rgs = [rg for rg in pf.row_groups if
         64            not(fastparquet.api.filter_out_stats(rg, filters, pf.schema)) and
         65            not(fastparquet.api.filter_out_cats(rg, filters))]
    

    C:\g\via\lib\dask154\dataframe\io\parquet.py in <listcomp>(.0)
         63     rgs = [rg for rg in pf.row_groups if
         64            not(fastparquet.api.filter_out_stats(rg, filters, pf.schema)) and
    ---> 65            not(fastparquet.api.filter_out_cats(rg, filters))]
         66 
         67     if index is False:
    

    C:\g\via\lib\fastparquet\api.py in filter_out_cats(rg, filters, sep)
        709             else:
        710                 v0 = val_to_num(v)
    --> 711             out = filter_val(op, val, v0, v0)
        712             if out is True:
        713                 return True
    

    C:\g\via\lib\fastparquet\api.py in filter_val(op, val, vmin, vmax)
        735         if op in ['==', '>='] and val > vmax:
        736             return True
    --> 737         if op == '>' and val >= vmax:
        738             return True
        739         if op == 'in' and min(val) > vmax:
    

    TypeError: unorderable types: int() >= str()

**Comment by martindurant** [comment]
I finally have a windows system up and running, I will be working on this and get back to you.

**Comment by apiszcz** [comment]
Great, thanks. NOTE: Mr. Rocklin has the same post for DASK.

**Comment by martindurant** [comment]
Can you try again with #232 ?

**Comment by apiszcz** [comment]
Getting a build error, io.h.

**Comment by martindurant** [comment]
Believed fixed since #271
